### PR TITLE
Removed unused code, tests for listeners

### PR DESCRIPTION
**Pull request by bartekgorny**
This PR addresses #711, though in a bit different way: the code in question had an unclear purpose and since 2014 was not used at all, so it could be safely removed. I added some tests to check whether listeners are started/stopped properly.

One notable change in management api: before, stopping listeners cause them to be erased from config, so stop and start were not symmetric - starting them again required running ejabberd_config:reload_local.

### CODE REVIEW

File: apps/ejabberd/test/ejabberd_listener_SUITE.erl [hunk 1]
```diff
@@ -2,9 +2,19 @@
...
+
+-define(DEFAULT_PORTS, [5222, 5280, 5269]).
+-define(ALT_PORTS, [5222, 5280, 5296]). %% yes it is different
+
+
 all() ->
-    [tcp_socket_is_started_with_default_backlog,
-     tcp_socket_is_started_with_options].
+    [tcp_start_stop_reload].
+%%    [tcp_socket_is_started_with_default_backlog,
```
Comment (michalwski) [inline apps/ejabberd/test/ejabberd_listener_SUITE.erl]: Why did you disable these tests?

File: apps/ejabberd/test/ejabberd_listener_SUITE_data/ejabberd.alt.cfg [hunk 2]
```diff
@@ -0,0 +1,745 @@
+%%%
```
Comment (michalwski) [inline apps/ejabberd/test/ejabberd_listener_SUITE_data/ejabberd.alt.cfg]: I know we did it before for other ejabberd.cfg used in tests, but I think it'd be better to remove all the commented lines, for the sake of readability.

File: apps/ejabberd/test/ejabberd_listener_SUITE.erl (hunk 1 above)
Comment (bartekgorny) [inline apps/ejabberd/test/ejabberd_listener_SUITE.erl]: ooops... just to make it run faster locally, and forgot to reenable them. Fortunately you are on guard :)

**Comment by michalwski** [comment]
Could you please modify all new file `ejabberd.*.cfg` by removing commented lines? This lines are not necessary for tests and makes the diff needlessly big.

**Comment by michalwski** [comment]
Ok, I removed these lines on my own. Also I removed ejabberd.cfg file as this one is produces from other config files and there is no need to keep it in repo.

**Pull request merged**